app/models/rubric_criterion.rb

```diff
@@ -38,6 +39,12 @@ def validate_max_mark
     errors.add(:max_mark, 'Max mark of rubric criterion should not be greater than max level mark')
   end
 
+  def only_update_if_results_unreleased
+    return if self.marks.empty?
+    return unless self.marks[0].result.released_to_students
```

Comment:
Checking the first mark isn't enough, there might be a mixture of released and unreleased results. Use a join on `self.marks` and results and filter by `released_to_students`. See https://guides.rubyonrails.org/active_record_querying.html#specifying-conditions-on-the-joined-tables.